cassidy-emily-ndvi.py

```diff
@@ -0,0 +1,634 @@
+            ndvi_array=ndvi_xr, folder_path=folder,
+            crop_bound=crop_bound, masked_values=all_masked_values)
+        
+        ndvi_mean = ndvi_clean_crop.mean(skipna=True)
+        ndvi_mean_value = ndvi_mean.item()
+        
+        path_components = folder.split(os.sep)
+        site = path_components[2]
+        file_string = path_components[4]
+        date = file_string[10:18]
+        date_time = datetime.strptime(date, '%Y%m%d').strftime('%m/%d/%Y')
+        ndvi_list.append([site, date_time, ndvi_mean_value])
+
+ndvi_df = pd.DataFrame(ndvi_list, columns=['site', 'date', 'mean_ndvi'])
+ndvi_df['date'] = pd.to_datetime(ndvi_df['date'])
+ndvi_df.set_index("date", inplace = True)
+ndvi_df
+
+
+# In[15]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points +=2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points +=2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points +=3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points +=3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# In[16]:
+
+
+# Add only the plot code to this cell
+
+# This is the final figure of mean NDVI
+# for both sites across the year
+# with data cleaned to deal with clouds
+
+f, ax = plt.subplots(figsize=(14, 9))
+for s, df in ndvi_df.dropna().groupby('site'):
+    ax.plot(df['mean_ndvi'], 'o-',  label=s)
+    
+ax.set(title="Mean Normalized Vegetation Difference Index (NDVI) for two NEON sites Over One Year",
+       xlabel='Date',
+       ylabel='NDVI Mean')
+plt.legend(bbox_to_anchor=(0.98,0.98))
+
+
```

Comment:
@emilyscassidy consider using the [suptitle() method](https://www.geeksforgeeks.org/matplotlib-figure-figure-suptitle-in-python/) from matplotlib as it will add a centered title to the figure. The `ax.set(title=)` would then become a subtitle and you could state which year. I'd also recommend stating which two sites being depicted in the title somewhere.